src/Compilers/Core/Portable/Generated/Operations.Generated.cs

```diff
@@ -3345,57 +3347,38 @@ public interface IWithOperation : IOperation
     #endregion
 
     #region Implementations
-    internal abstract partial class BaseBlockOperation : OperationOld, IBlockOperation
+    #nullable enable
+    internal sealed partial class BlockOperation : Operation, IBlockOperation
     {
-        internal BaseBlockOperation(ImmutableArray<ILocalSymbol> locals, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, ConstantValue constantValue, bool isImplicit)
-            : base(OperationKind.Block, semanticModel, syntax, type, constantValue, isImplicit)
+        private IEnumerable<IOperation>? _lazyChildren;
+        internal BlockOperation(ImmutableArray<IOperation> operations, ImmutableArray<ILocalSymbol> locals, SemanticModel? semanticModel, SyntaxNode syntax, bool isImplicit)
+            : base(semanticModel, syntax, isImplicit)
         {
+            Operations = SetParentOperation(operations, this);
             Locals = locals;
         }
-        public abstract ImmutableArray<IOperation> Operations { get; }
+        public ImmutableArray<IOperation> Operations { get; }
         public ImmutableArray<ILocalSymbol> Locals { get; }
         public override IEnumerable<IOperation> Children
         {
             get
             {
-                foreach (var child in Operations)
+                if (_lazyChildren is null)
                 {
-                    if (child is object) yield return child;
+                    var builder = ArrayBuilder<IOperation>.GetInstance(1);
+                    if (!Operations.IsEmpty) builder.AddRange(Operations);
+                    Interlocked.CompareExchange(ref _lazyChildren, builder.ToImmutableAndFree(), null);
```

Comment:
Given that we're checking for `IsEmpty` here why allocate the builder at all if it's empty? Seems more efficient here to just use `ImmutableArray<IOperation>.Empty` in that case.